vcd/resource_vcd_edgegateway_test.go

```diff
@@ -0,0 +1,155 @@
+// +build gateway ALL functional
+
+package vcd
+
+import (
+	"testing"
+
+	"github.com/hashicorp/terraform/helper/resource"
+	"github.com/hashicorp/terraform/terraform"
+)
+
+func TestAccVcdEdgeGatewayBasic(t *testing.T) {
+	var edgeGatewayName string = "TestEdgeGatewayBasic"
+	var edgeGatewayVcdName string = "test_edge_gateway_basic"
+
+	// String map to fill the template
+	var params = StringMap{
+		"Org":             testConfig.VCD.Org,
+		"Vdc":             testConfig.VCD.Vdc,
+		"EdgeGateway":     edgeGatewayName,
+		"EdgeGatewayVcd":  edgeGatewayVcdName,
+		"ExternalNetwork": testConfig.Networking.ExternalNetwork,
+		"Tags":            "gateway",
+	}
+	configText := templateFill(testAccEdgeGatewayBasic, params)
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckVcdEdgeGatewayDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"vcd_edgegateway."+edgeGatewayName, "default_gateway", testConfig.Networking.ExternalNetwork),
+				),
+			},
+		},
+	})
+}
+
+func TestAccVcdEdgeGatewayComplex(t *testing.T) {
+	var (
+		edgeGatewayName       string = "TestEdgeGatewayComplex"
+		edgeGatewayVcdName    string = "test_edge_gateway_complex"
+		newExternalNetwork    string = "TestExternalNetwork"
+		newExternalNetworkVcd string = "test_external_network"
+	)
+
+	// String map to fill the template
+	var params = StringMap{
+		"Org":                   testConfig.VCD.Org,
+		"Vdc":                   testConfig.VCD.Vdc,
+		"EdgeGateway":           edgeGatewayName,
+		"EdgeGatewayVcd":        edgeGatewayVcdName,
+		"ExternalNetwork":       testConfig.Networking.ExternalNetwork,
+		"Tags":                  "gateway",
+		"NewExternalNetwork":    newExternalNetwork,
+		"NewExternalNetworkVcd": newExternalNetworkVcd,
+		"Type":                  testConfig.Networking.ExternalNetworkPortGroupType,
+		"PortGroup":             testConfig.Networking.ExternalNetworkPortGroup,
+		"Vcenter":               testConfig.Networking.Vcenter,
+	}
+	configText := templateFill(testAccEdgeGatewayComplex, params)
+	if vcdShortTest {
+		t.Skip(acceptanceTestsSkipped)
+		return
+	}
+	debugPrintf("#[DEBUG] CONFIGURATION: %s", configText)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckVcdEdgeGatewayDestroy,
+		Steps: []resource.TestStep{
+			resource.TestStep{
+				Config: configText,
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(
+						"vcd_edgegateway."+edgeGatewayName, "default_gateway", newExternalNetworkVcd),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckVcdEdgeGatewayDestroy(s *terraform.State) error {
```

Comment:
I would like to see check and edge gtw isn't anymore in vCD

Reply:
done